cfme/cloud/keypairs.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+import cfme.fixtures.pytest_selenium as sel
+from cfme.common import SummaryMixin, Taggable
+from cfme.web_ui import (Quadicon, flash, Form, Input, form_buttons, fill, summary_title,
+    breadcrumbs, AngularSelect)
+from functools import partial
+from cfme.web_ui import toolbar as tb
+from utils.wait import wait_for
+
+
+cfg_btn = partial(tb.select, "Configuration")
+keypairs_form = Form(
+    fields=[
+        ('name', Input("name")),
+        ('public_key', Input("public_key")),
+        AngularSelect('ems_id'),
+        ('save_button', form_buttons.save)
+    ])
+
+
+class KeyPair(Taggable, SummaryMixin):
+    """ Model of KeyPairs
+
+    Args:
+        name: Name of Keypairs.
+    """
+
+    def __init__(self, name=None):
+        self.name = name
+
+    def _on_detail_page(self):
+        title = '{} (Summary)'.format(self.name)
+        return breadcrumbs() == ['Key Pairs', title] and summary_title() == title
+
+    def load_details(self, refresh=False):
+        if not self._on_detail_page():
+            self.navigate(detail=True)
+        elif refresh:
+            tb.refresh()
+
+    def delete(self):
+        sel.force_navigate('clouds_key_pairs', context={'keypairs': self})
+        cfg_btn("Remove this Keypairs from the VMDB", invokes_alert=True)
+        sel.handle_alert(cancel=False)
+        flash.assert_success_message('The selected Keypairs was deleted')
+
+    def nav_to_parameters_link(self):
+        self.summary.relationship.instances.click()
+
+    def wait_for_delete(self):
+        sel.force_navigate("clouds_key_pairs")
+        quad = Quadicon(self.name, 'keypairs')
+        wait_for(lambda: not sel.is_displayed(quad), fail_condition=False,
+            message="Wait keypairs to disappear", num_sec=500, fail_func=sel.refresh)
+
+    def add(self, cancel=False, refresh=True, refresh_timeout=120):
```

Comment:
And I don't think create for keypair should refresh, I guess you should remove the refresh funcionality from this method.